src/network/networkctl-config-file.c

```diff
@@ -482,6 +551,48 @@ int verb_edit(int argc, char *argv[], void *userdata) {
         return reload_daemons(reload);
 }
 
+static int cat_files_by_link_config(const char *link_config, sd_netlink **rtnl) {
+        _cleanup_free_ char *ifname = NULL, *path = NULL;
+        _cleanup_strv_free_ char **dropins = NULL;
+        LinkConfigType type;
+        int r;
+
+        assert(link_config);
+        assert(rtnl);
+
+        r = parse_link_config(link_config, &ifname, &type);
+        if (r < 0)
+                return r;
+
+        if (type == _CONFIG_MAX) {
+                for (LinkConfigType i = 0; i < _CONFIG_MAX; i++) {
+                        r = get_config_files_by_link_config(ifname, i,
+                                                            /* ignore_missing = */ true,
+                                                            rtnl,
+                                                            &path, &dropins);
+                        if (IN_SET(r, -ENOENT, -ESRCH))
+                                continue;
+                        if (r < 0)
+                                return r;
+
+                        r = cat_files(path, dropins, /* flags = */ CAT_FORMAT_HAS_SECTIONS);
+                        if (r < 0)
+                                return r;
```

Comment:
Not a matter, but I'd like to split out this as `cat_files_by_type()` or so. Then, we can reuse that in the below. Just a suggestion, feel free to ignore if you do not like.